src/parser.ts

```diff
@@ -442,6 +437,9 @@ function getNameText(
             return `set ${nameText}`;
         }
     }
+    if (ts.isPrivateIdentifier(name)) {
```

Comment:
There's a little weirdness in TypeScript that I didn't get to the bottom of: the `PropertyName` type includes `PrivateIdentifier`:

```ts
// typescript.d.ts
export type PropertyName = Identifier | StringLiteral | NumericLiteral | ComputedPropertyName | PrivateIdentifier;
```

But the `isPropertyName` function doesn't seem to agree:

```js
// typescript.js
function isPropertyName(node) {
    var kind = node.kind;
    return kind === 75 /* Identifier */
        || kind === 10 /* StringLiteral */
        || kind === 8 /* NumericLiteral */
        || kind === 154 /* ComputedPropertyName */;
}
```

If this were resolved this might simplify.